command/ca/provisioner/add.go

```diff
@@ -588,6 +611,73 @@ func addX5CProvisioner(ctx *cli.Context, name string, provMap map[string]bool) (
 	return
 }
 
+// addK8sSAProvisioner returns a provisioner list containing a kubernetes
+// service account provisioner.
+// NOTE: step-ca currently only supports one k8sSA provisioner (because we do
+// not have a good way of distinguishing between tokens), therefore w/e `name`
+// is entered by the user will be overwritten by a default value.
+func addK8sSAProvisioner(ctx *cli.Context, _ string, provMap map[string]bool) (list provisioner.List, err error) {
+	// Overwriting user input name with default.
+	name := provisioner.K8sSAName
+
+	pemKeysF := ctx.String("pem-keys")
+	if len(pemKeysF) == 0 {
+		return nil, errs.RequiredWithFlagValue(ctx, "type", "k8sSA", "pem-keys")
+	}
+
+	pemKeysB, err := ioutil.ReadFile(pemKeysF)
+	if err != nil {
+		return nil, errors.Wrap(err, "error reading pem keys")
+	}
+
+	var (
+		block   *pem.Block
+		rest    = pemKeysB
+		pemKeys = []interface{}{}
+	)
+	for rest != nil {
+		block, rest = pem.Decode(rest)
+		if block == nil {
+			break
+		}
+		key, err := pemutil.ParseKey(pem.EncodeToMemory(block))
+		if err != nil {
+			return nil, errors.Wrapf(err, "error parsing public key from %s", pemKeysF)
+		}
+		switch q := key.(type) {
+		case *rsa.PublicKey, *ecdsa.PublicKey, ed25519.PublicKey:
+		default:
+			return nil, errors.Errorf("Unexpected public key type %T in %s", q, pemKeysF)
+		}
+		pemKeys = append(pemKeys, key)
+	}
+
+	var pubKeyBytes []byte
+	for _, k := range pemKeys {
+		blk, err := pemutil.Serialize(k)
+		if err != nil {
+			return nil, errors.Wrap(err, "error serializing pem key")
+		}
+		pubKeyBytes = append(pubKeyBytes, pem.EncodeToMemory(blk)...)
+	}
```

Comment:
Is there any point in deserializing and serializing the keys?

Reply:
In most cases, the serialization will be equivalent to `pemKeysB`.

Reply:
Gives us the opportunity to fail when you're adding the provisioner, rather than failing when you attempt to sighup/restart the CA. I'd say it's nice to have slightly more assurance that the CA won't error out on restart.

Reply:
sounds good